src/Controller/Action/BulkEditProductsAction.php

```diff
@@ -25,27 +27,41 @@ final class BulkEditProductsAction
 
     private Environment $twig;
 
+    private RouterInterface $router;
+
     public function __construct(
         ProductRepositoryInterface $productRepository,
         ProductVariantRepositoryInterface $productVariantRepository,
         ChannelRepositoryInterface $channelRepository,
-        Environment $twig
+        Environment $twig,
+        RouterInterface $router
     ) {
         $this->productRepository = $productRepository;
         $this->productVariantRepository = $productVariantRepository;
         $this->channelRepository = $channelRepository;
         $this->twig = $twig;
+        $this->router = $router;
     }
 
     public function __invoke(Request $request): Response
     {
         /** @var ChannelInterface[] $channels */
         $channels = $this->channelRepository->findAll();
+        Assert::minCount($channels, 1);
 
         if (!$request->query->has('channelCode')) {
-            return new RedirectResponse($request->getUri() . '&channelCode=' . $channels[0]->getCode());
+            $channelAwareUrl = $request->getUri();
+            $channelAwareUrl .= '&channelCode=' . $channels[0]->getCode();
+
+            return new RedirectResponse($channelAwareUrl);
```

Comment:
Why was this oneliner converted into four lines?

Reply:
To have less width :)